File: 007_Reverse_Integer/solution.py
class Solution(object):
    def reverse(self, x):
        """
        :type x: int
        :rtype: int
        """
        flag = cmp(x, 0)
        x *= flag
        
        result = 0
        while x != 0:
            k = x % 10
            result = result * 10 + k
            if result >= 2**31:
                return 0
            x //= 10
        return result * flag
    
    # Runtime: 42ms

File: 007_Reverse_Integer/Solution.java
public class Solution {
    public int reverse(int x) {
        int result = 0;
        while (x != 0) {
            int k = x % 10;
            int newResult = result * 10 + k;
            if ((newResult - k) / 10 != result)
                return 0;
            result = newResult;
            x /= 10;
        }
        return result;
    }
}

// Runtime: 43ms